### PR TITLE
chore(phlare/pyroscope): rename phlare to pyroscope

### PR DESCRIPTION
**Pull request by korniltsev**
#### PR Description

As part of phlare and pyroscope merging we are removing references to phlare, keeping it pyroscope.

- renamed `component/phlare` package to `component/pyroscope`
- renamed documentation

#### Which issue(s) this PR fixes
https://github.com/grafana/pyroscope-squad/issues/20

#### Notes to the Reviewer

Should we update upgrade guide?

#### PR Checklist

- [x] CHANGELOG updated
- [x] Documentation added
- [ ] Tests updated

### CODE REVIEW

**Review by rfratto — COMMENTED** [review]
LGTM but I'd like to check with @cyriltovena that we're ready for the components to be renamed.

File: CHANGELOG.md [hunk 1]
```diff
@@ -19,6 +19,8 @@ Main (unreleased)
 - Upgrade the embedded windows_exporter to commit 79781c6. (@jkroepke)
 
 - Prometheus exporters in Flow mode now set the `instance` label to a value similar to the one they used to have in Static mode (<hostname> by default, customized by some integrations). (@jcreixell)
+ 
+- `phlare.scrape` and `phlare.write` have been renamed to `pyroscope.scrape` and `pyroscope.scrape`. (@korniltsev)
```
[the remark this answers is not in the document]
Comment (rfratto) [reply CHANGELOG.md:23]: (Breaking changes also would need to be documented in the upgrade guide: https://github.com/grafana/agent/blob/main/docs/sources/flow/upgrade-guide.md)

[the remark this answers is not in the document]
Comment (korniltsev) [reply CHANGELOG.md:23]: This item seems to be under "Breaking changes" section, so it is everything correct here? right?

I will update upgrade guide

[the remark this answers is not in the document]
Comment (rfratto) [reply CHANGELOG.md:23]: > This item seems to be under "Breaking changes" section, so it is everything correct here? right?

Yes, sorry, I got confused by some of the other items 🤦 The only thing to do here is to update the upgrade guide.

**Review by clayton-cornell — COMMENTED** [review]
From the docs side it looks OK.

**Review by rfratto — APPROVED** [review]
LGTM, thank you!

**Pull request merged**